### PR TITLE
PE-1336: Setup dev stack for Inferno

### PR DESCRIPTION
**Pull request by matibat**
- Adds Yarn + PnP
- Adds typescript
- Copies the editor-config, prettier and linting rules from CLI

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+		"format": "prettier --write \"src/**/*.ts\"",
+		"lint": "eslint . --ext .ts",
+		"lintfix": "eslint . --ext .ts --fix",
+		"typecheck": "tsc --noemit",
+		"build": "yarn clean && tsc --project ./tsconfig.prod.json",
+		"dev": "yarn clean && tsc --project ./tsconfig.prod.json -w"
+	},
+	"devDependencies": {
+		"@types/eslint": "^8",
+		"@types/rimraf": "^3",
```
Comment (karlprieb) [inline package.json]: why we need rimraf types?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+		"eslint": "^7.23.0",
+		"eslint-config-prettier": "^8.1.0",
+		"eslint-plugin-prettier": "latest",
+		"lint-staged": "^11.0.0",
+		"prettier": "^2.2.1",
+		"typescript": "^4.6.3"
+	},
+	"packageManager": "yarn@3.2.0",
+	"dependencies": {
+		"eslint": "^8.13.0",
```
Comment (karlprieb) [inline package.json]: why two eslint versions?

**Comment by karlprieb** [comment]
We're using Yarn v3.2 and not v2 :)

File: package.json (hunk 1 above)
Comment (matibat) [reply package.json]: ccc72f5bbdaaa17c26390700ad7c50a35ff2d6f8

File: package.json (hunk 2 above)
Comment (matibat) [reply package.json]: Nice catch ccc72f5bbdaaa17c26390700ad7c50a35ff2d6f8

**Review by karlprieb — DISMISSED** [review]

File: package.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+		"typecheck": "tsc --noemit",
+		"build": "yarn clean && tsc --project ./tsconfig.prod.json",
+		"dev": "yarn clean && tsc --project ./tsconfig.prod.json -w"
+	},
+	"devDependencies": {
+		"@yarnpkg/sdks": "^3.0.0-rc.2",
+		"eslint": "^8.13.0",
+		"eslint-config-prettier": "^8.1.0",
+		"eslint-plugin-prettier": "latest",
+		"lint-staged": "^11.0.0",
```
Comment (agsuy) [inline package.json:29]: lint-staged without husky? also you might want to setup a hook on commits so lint-staged runs like we have on https://github.com/ardriveapp/ardrive-cli/blob/master/.husky/pre-commit

**Review by agsuy — COMMENTED** [review]
Left a comment on git hooks.

**Review by karlprieb — DISMISSED** [review]

File: package.json (hunk 3 above)
Comment (matibat) [reply package.json:29]: I've added husky, but it seems I'm missing something because it's not working atm

**Review by karlprieb — APPROVED** [review]

**Pull request merged**